### PR TITLE
updated train and eval functions with loss metrics

### PR DESCRIPTION
**Pull request by TWintersww**

### CODE REVIEW

File: Super Resolution/Schrodinger Diffusion/models/models.py [hunk 1]
```diff
@@ -129,7 +134,7 @@ def __init__(self, ResNet = None, image_size=224, num_input_channels=3):
         self.layer3 = ResNet.layer3
         self.layer4 = ResNet.layer4
 
-        dummy_input = torch.randn(1, num_input_channels, self.input_image_size, self.input_image_size)
+        dummy_input = torch.randn(1, 3, self.input_image_size, self.input_image_size)
```
Comment (gagewrye) [inline Super Resolution/Schrodinger Diffusion/models/models.py:137]: It would be better not to hardcode parameters if you can help it

File: Super Resolution/Schrodinger Diffusion/models/models.py [hunk 2]
```diff
@@ -90,7 +91,9 @@ def forward(self, image: torch.tensor, diffuse: bool = True, return_encoding_onl
                 x = self.diffuser(x, step)
             else:
                 for i in range(1, self.num_timesteps + 1):
-                    time_step = torch.tensor([i], dtype=torch.float32)
+                    # instead of having time_step be a [1] length array of i, we might want it to be a [B] length array of i
```
Comment (gagewrye) [inline Super Resolution/Schrodinger Diffusion/models/models.py:94]: Yeah, we should probably support batch sizing here, that's what I'm assuming this means. We would need to check if the latent input is batched. For modularity, it would be best to check the shape of the latent space in the init, save it, and then compare the input to see if there is an added batch dim.

File: Super Resolution/Schrodinger Diffusion/i2sb/runner.py [hunk 3]
```diff
@@ -122,15 +125,20 @@ def __init__(self, opt):
...
             checkpoint = torch.load(opt.load, map_location="cpu")
             self.net.load_state_dict(checkpoint['net'])
             self.ema.load_state_dict(checkpoint["ema"])
+            print("Loaded 'net' and 'ema' from checkpoint path")
         self.net.to(opt.device)
         self.ema.to(opt.device)
         print(f"Built {opt.diffusion_type} Diffusion Model with {len(betas)} steps and {opt.beta_schedule} beta schedule!")
     
+    # train using algorithm 1 described in paper
     def train(self, opt, train_dataset, val_dataset):
```
Comment (gagewrye) [inline Super Resolution/Schrodinger Diffusion/i2sb/runner.py:142]: Do we have the capability to change the type of loss by passing in an opt parameter? It would be good to have the option to try out different types of loss functions.
